#exception =abnormal condition errors
# EG:
#
# a=int(input('enter a no'))
# b=int(input('enter a no'))
# div=a/b
# print(div)

# exception handling
# a=int(input('enter a no'))
# b=int(input('enter a no'))
# try:                                            #suspected code
#      div=a/b
#      print(div)
# except:                      #exception handling
#         print('divison by zero is not possible')
# else:   # execute when there is  no exception
#     print("iam in else block")
# finally: #execute both times    print("done")
#            print("done")



# a=int(input('enter a no'))
# b=int(input('enter a no'))
# try:                                            #suspected code
#      div=a/b
#      print(div)
# except:                      #exception handling
#         print('divison by zero is not possible')
# else:   # execute when there is  no exception
#     print("iam in else block")
# finally: #execute both times    print("done")
#          print("done")



#
# a=int(input('enter a no'))
# b=int(input('enter a no'))
# x=2
# try:                                            #suspected code
#
#      div=a/b
#      print(div)
#      print(x)
# except ZeroDivisionError:                      #exception handling
#         print('divison by zero is not possible')
# except NameError:
#        print('variable is not assigned')






# try:                                            #suspected code
#     a = int(input('enter a no'))
#     b = int(input('enter a no'))
#     x=54
#     div=a/b
#     print(div)
#     print(x)
# except ZeroDivisionError:                      #exception handling
#         print('divison by zero is not possible')
# except NameError:
#        print('variable is not assigned')
# except ValueError:
#     print('expected a numeric value not letter and others')


# try:                                            #suspected code
#     a = int(input('enter a no'))
#     b = int(input('enter a no'))
#     x=54
#     div=a/b
#     print(div)
#     print(x)
# except ZeroDivisionError:                      #exception handling
#         print('divison by zero is not possible')
# except NameError:
#        print('variable is not assigned')
# except :
#     print('expected a numeric value not letter and others')



